src/node_process_methods.cc

```diff
@@ -157,14 +157,20 @@ static void Kill(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(err);
 }
 
-static void MemoryUsage(const FunctionCallbackInfo<Value>& args) {
+static void Rss(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
   size_t rss;
   int err = uv_resident_set_memory(&rss);
   if (err)
     return env->ThrowUVException(err, "uv_resident_set_memory");
 
+  args.GetReturnValue().Set(static_cast<double>(rss));
+}
```

Comment:
Given what this is doing, and given that performance is the motivator here, this would be a good candidate for the new v8 fast api calls capability. (see the implementation of the fast hrtime implementation in node_process_methods, @devsnek may be able to help with pointers there). 

I'm also wondering if the cast to double is really needed here? Since this is providing a new API, why not just go ahead and return this as a bigint?

Reply:
I try to understand the code of the `FastHrtime` but I struggle to understand how it works.

I cast it to `double` because it's the only big enough authorized type for this templated function

```
../deps/v8/include/v8.h:11159:6: note: candidate: ‘void v8::ReturnValue<T>::Set(bool) [with T = v8::Value]’
11159 | void ReturnValue<T>::Set(bool value) {
      |      ^~~~~~~~~~~~~~
../deps/v8/include/v8.h:11130:6: note: candidate: ‘void v8::ReturnValue<T>::Set(double) [with T = v8::Value]’
11130 | void ReturnValue<T>::Set(double i) {
      |      ^~~~~~~~~~~~~~
../deps/v8/include/v8.h:11136:6: note: candidate: ‘void v8::ReturnValue<T>::Set(int32_t) [with T = v8::Value; int32_t = int]’
11136 | void ReturnValue<T>::Set(int32_t i) {
      |      ^~~~~~~~~~~~~~
../deps/v8/include/v8.h:11147:6: note: candidate: ‘void v8::ReturnValue<T>::Set(uint32_t) [with T = v8::Value; uint32_t = unsigned int]’
11147 | void ReturnValue<T>::Set(uint32_t i) {
      |      ^~~~~~~~~~~~~~
```

Reply:
fast call return values and floats are still WIP so this might not be a good candidate*yet* but when they are supported better it would definitely be worth migrating.